packages/three-vrm-core/src/expressions/VRMExpressionMaterialColorBind.ts

```diff
@@ -75,84 +96,119 @@ export class VRMExpressionMaterialColorBind implements VRMExpressionBind {
+      }
     }
   }
 
-  public applyWeight(weight: number): void {
-    if (this._state == null) {
-      // warning is already emitted in constructor
-      return;
-    }
+  public clearAppliedWeight(): void {
+    const { color, alpha } = this._state;
 
-    const { propertyName, deltaValue } = this._state;
+    if (color != null) {
+      const { propertyName, initialValue } = color;
 
-    const target = (this.material as any)[propertyName] as THREE.Color;
-    if (target === undefined) {
-      return;
-    } // TODO: we should kick this at `addMaterialValue`
+      const target = (this.material as any)[propertyName] as THREE.Color;
+      if (target != undefined) {
+        target.copy(initialValue);
+      }
+    }
 
-    target.add(_color.copy(deltaValue).multiplyScalar(weight));
+    if (alpha != null) {
+      const { propertyName, initialValue } = alpha;
 
-    if (typeof (this.material as any).shouldApplyUniforms === 'boolean') {
-      (this.material as any).shouldApplyUniforms = true;
+      const target = (this.material as any)[propertyName] as number;
+      if (target != undefined) {
+        ((this.material as any)[propertyName] as number) = initialValue;
+      }
     }
   }
 
-  public clearAppliedWeight(): void {
-    if (this._state == null) {
-      // warning is already emitted in constructor
-      return;
+  private _initColorBindState(): ColorBindState | null {
+    const { material, type, targetValue } = this;
+
+    const propertyNameMap = this._getPropertyNameMap();
+    const propertyName = propertyNameMap?.[type]?.[0] ?? null;
+
+    if (propertyName == null) {
+      console.warn(
+        `Tried to add a material color bind to the material ${
+          material.name ?? '(no name)'
+        }, the type ${type} but the material or the type is not supported.`,
+      );
+
+      return null;
     }
 
-    const { propertyName, initialValue } = this._state;
+    const target = (material as any)[propertyName] as THREE.Color;
+
+    const initialValue = target.clone();
+
+    // 負の値を保持するためにColor.subを使わずに差分を計算する
+    const deltaValue = new THREE.Color(
+      targetValue.r - initialValue.r,
+      targetValue.g - initialValue.g,
+      targetValue.b - initialValue.b,
+    );
 
-    const target = (this.material as any)[propertyName] as THREE.Color;
-    if (target === undefined) {
-      return;
-    } // TODO: we should kick this at `addMaterialValue`
+    return { propertyName, initialValue, deltaValue };
+  }
+
+  private _initAlphaBindState(): AlphaBindState | null {
+    const { material, type, targetAlpha } = this;
 
-    target.copy(initialValue);
+    const propertyNameMap = this._getPropertyNameMap();
+    const propertyName = propertyNameMap?.[type]?.[1] ?? null;
 
-    if (typeof (this.material as any).shouldApplyUniforms === 'boolean') {
-      (this.material as any).shouldApplyUniforms = true;
+    if (propertyName == null) {
+      return null;
```

Comment:
should we warn for AlphaBind if not supported too?

Reply:
No, we don't have to.
I wondered if we could check if the target alpha == 1 when the bind is for types that doesn't support alpha, but the model, just in case. How do you think?

Reply:
IMO it's a good to have but not necessary. It might be confusing when changing alpha for a wrong type but nothing happens.

Reply:
Addressed @ https://github.com/pixiv/three-vrm/pull/1323/commits/2dded77dbe63ed3323f691c219ca92e6f72ca8be